### ISSUE TITLE
Inconsistent definitions for velocity profiles

### ISSUE THREAD

**Issue by ischoegl** [bryanwweber/CanSen#45]
In `profiles.py`, `VolumeProfiles` outputs dV/dt as velocity, whereas `ICEngineProfile` calculates the actual piston velocity dx/dt. I stumbled across this issue looking at PyKED's [RCM example](https://github.com/ischoegl/PyKED/blob/master/docs/rcm-example.ipynb). Presumably both classes are thought as plugins for cantera's `Wall` objects, which uses the wall velocity as input (in m/s).  where for the RCM _the area has to be set to 1_: otherwise, the velocity will not be calculated correctly. 

I am not certain to what extent `CanSen` is under development, but the PyKED example draws from here, and certainly uses incorrect units. Curiously, neither bore diameter nor piston area are stored in the current PyKED format, so it is not an easy fix.

**Comment by bryanwweber** [comment]
The input volume profile is normalized in the `VolumeProfile` case, so the default area of 1 m**2 is fine, with the exception that any extensive quantities (e.g. total heat release) are not meaningful. However intensive properties like temperature and pressure (which are more important for RCM experiments) match just fine.

**Comment by ischoegl** [comment]
OK, I guess I overlooked `self.volume = np.array(keywords['vproVol'])/keywords['vproVol'][0]`. Definitions are still not consistent though.

**Comment by ischoegl** [comment]
Closing here as I posted a similar comment for PyKED.

**Comment by bryanwweber** [comment]
I'm going to reopen this so I remember to document more clearly what is happening here.

**Comment by ischoegl** [comment]
As an aside, I stumbled across this while working on a cantera PR that targets a more flexible handling of Wall and FlowDevice objects (Cantera/cantera#632). For this, I decided to implement an alternative wall object (`ReciprocatingPiston`) as an illustration. While this is not my primary research rationale, Cantera's `ic_engine.py` is hard to follow and unfortunately not very useful without good working knowledge of python (i.e. a typical student in my combustion class). I just force-pushed a version that uses variable names that are consistent with `ICEngineProfile`. As long as the PR is of interest, I can easily include another class that does the same for RCM's (with the benefit that a native C++ implementation is considerably faster). Any feedback is welcome ...

**Comment by bryanwweber** [comment]
We're having an internal debate at Cantera about where the boundary is for things that belong in "core" Cantera vs. are application specific enough that they should be in a separate package (perhaps also maintained by the Cantera org, or at least loosely affiliated). Please feel free to push that example/RCM class, but I suspect that we'll have to have a discussion about whether including that or the `ICEngineProfile` is something we want to include at all in "core" Cantera, or if it should be separate. First we'll need to define what "core" Cantera even is... Anyhow, with the end of the semester right now, it will probably be a few weeks until any of us can take a good look at the changes and start that conversation.

**Comment by ischoegl** [comment]
Understood (I have to grade myself). Compilation does take a while, as everything is currently being handled by a single library. Would it make sense to split into core / 0D / 1D libraries? Compiling against cantera is not overly complicated (see, e.g. a template I created in [ischoegl/ctapp](https://github.com/ischoegl/ctapp)), but I'd argue that a couple of application oriented objects would benefit the average user. Also, especially with conda/jupyter, cantera is becoming extremely useful in teaching.

**Comment by bryanwweber** [comment]
That might be a good way to do it. I'm not sure what's the best place to have these high level architectural conversations, and this is something we've only started discussing ourselves. If you email me and Ray and Steven, that might be the best way to start a discussion, but eventually I imagine we'd want to move it somewhere public for documentation purposes.

**Comment by ischoegl** [comment]
Per our discussion, I added a class to my PR that implements an equivalent of `VolumeProfiles` in the C++ layer. I am about to send the e-mail you suggested: besides you, Ray and Steven, are there any other people I should include?

**Comment by bryanwweber** [comment]
No, I think the three of us are the most appropriate to start the discussion. As I said, we'll move it somewhere public eventually, but let's start small.
